### PR TITLE
Added DomainPublishedMatcher and minor refactor of ProjectSpec

### PR DESCRIPTION
**Pull request by gorkalaucirica**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | no
| License       | MIT
| Doc PR        | no

### CODE REVIEW

File: src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * Usage:
+ *    $this->shouldHavePublished(EventPublished::class);
+ * 
+ * @package Kreta\SharedKernel\Tests\Matchers
+ */
+class DomainPublishedMatcher extends BasicMatcher
+{
+    protected function matches($subject, array $arguments)
+    {
+        /** @var AggregateRoot $subject */
```
Comment (benatespina) [inline src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php]: You should avoid PHPDocs

File: src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function getFailureException($name, $subject, array $arguments)
+    {
+        /* @var AggregateRoot $subject */
```
Comment (benatespina) [inline src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php]: You should avoid PHPDocs

Comment (gorkalaucirica) [inline src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php]: Just for help in the IDE

File: src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php (hunk 1 above)
Comment (gorkalaucirica) [inline src/Kreta/SharedKernel/Tests/Matchers/DomainPublishedMatcher.php]: Just for help in the IDE

**Pull request merged**